bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -337,11 +337,30 @@ private void disposeWebsocketPollingJob() {
     private void pollingWebsocketJob() {
         switch (yioRemoteDockActualStatus) {
             case AUTHENTICATION_COMPLETE:
-                if (getAndResetHeartbeat()) {
-                    updateChannelString(GROUP_OUTPUT, STATUS_STRING_CHANNEL,
-                            irCodeReceivedHandler.getCode() + irCodeReceivedHandler.getFormat());
-                    logger.debug("heartBeat ok");
+                if (resetHeartbeat()) {
                     sendMessage(YioRemoteMessages.HEARTBEAT_MESSAGE, "");
+                    yioRemoteDockActualStatus = YioRemoteDockHandleStatus.CHECK_PONG;
+                } else {
+                    yioRemoteDockActualStatus = YioRemoteDockHandleStatus.COMMUNICATION_ERROR;
+                }
+                break;
+            case SEND_PING:
+                if (resetHeartbeat()) {
+                    sendMessage(YioRemoteMessages.HEARTBEAT_MESSAGE, "");
+                    yioRemoteDockActualStatus = YioRemoteDockHandleStatus.CHECK_PONG;
+                } else {
+                    yioRemoteDockActualStatus = YioRemoteDockHandleStatus.COMMUNICATION_ERROR;
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Connection lost no ping from YIO DOCK");
+                    disposeWebsocketPollingJob();
+                    reconnectWebsocket();
```

Comment:
`resetHeartbeat()` always returns true now, so this code will never get called anymore...

Reply:
done